### ISSUE TITLE
Generate docs from ocean-lib

### ISSUE THREAD

**Issue by mariacarmina** [oceanprotocol/ocean.py#1433]
Generate docs from ocean-lib important modules comments & doc strings.

**Comment by trentmc** [comment]
You know we did this already, right? Via Akshay.

And there was a huge amount of work to make it pretty. And in the end, it was pretty.

So if you're gonna do that, you should leverage all the past work done for this.

**Comment by mariacarmina** [comment]
I am sorry maybe I missed that. I didn't know that we have docs generated by our Python doc strings or Python comments. Can you share with me Akshay's work so I can iterate on that please?
Thank you for letting me know!

**Comment by trentmc** [comment]
About the previous docs:
- First, know that it probably took 6+ man-months of work to make it look good. There were a lot of gotchas to tune it, across different chunks of code and software stacks. Not easy at all. 
- The first thing to do is: play with what the final result looked like. For that, just revive an older version of docs.oceanprotocol.com (perhaps 3-6 months ago).
- Then dig around oceanprotocol org to see the repos that made this. (There were a few). 
- Pros: look good, thorough, "automated"
- Cons: not very readable, super-hard to maintain, hard to make updates

---

Tbh I don't think auto-generating is the way to go. It might look like less work to get "something". But better is to have shorter docs that are well-targeted and explain well, rather than super-thorough but without good guidance on what matters. 

Good example docs, at a similar level of maturity to ocean.py, are:
- Zama. https://docs.zama.ai/concrete
- Brownie. https://eth-brownie.readthedocs.io/en/stable/

Zama docs were built with gitbook. Much simpler flow to update compared to our current docs repo. IIRC Ana & others were exploring that. I'm not sure the status.

I _strongly_ recommend to go a route to have our docs look something like Zama or Brownie, eg via gitbook. The outcome will be way better, and way easier to continually involve.

**Comment by mariacarmina** [comment]
> About the previous docs:
> 
> * First, know that it probably took 6+ man-months of work to make it look good. There were a lot of gotchas to tune it, across different chunks of code and software stacks. Not easy at all.
> * The first thing to do is: play with what the final result looked like. For that, just revive an older version of docs.oceanprotocol.com (perhaps 3-6 months ago).
> * Then dig around oceanprotocol org to see the repos that made this. (There were a few).
> * Pros: look good, thorough, "automated"
> * Cons: not very readable, super-hard to maintain, hard to make updates
> 
> Tbh I don't think auto-generating is the way to go. It might look like less work to get "something". But better is to have shorter docs that are well-targeted and explain well, rather than super-thorough but without good guidance on what matters.
> 
> Good example docs, at a similar level of maturity to ocean.py, are:
> 
> * Zama. https://docs.zama.ai/concrete
> * Brownie. https://eth-brownie.readthedocs.io/en/stable/
> 
> Zama docs were built with gitbook. Much simpler flow to update compared to our current docs repo. IIRC Ana & others were exploring that. I'm not sure the status.
> 
> I _strongly_ recommend to go a route to have our docs look something like Zama or Brownie, eg via gitbook. The outcome will be way better, and way easier to continually involve.

I understand your point and thank you for putting the resources here & giving more context. I'll tag @AnaLoznianu, maybe she'll provide an opinion as well.

**Comment by AnaLoznianu** [comment]
@trentmc We are currently working on upgrading our docs and having better documentation for our components. Ocean.py and ocean.js are part of this.
I don't envision importing the generated documentation in the docs repo but having a few pages detailing some of the functionalities with references to the autogenerated content. 
I know it's not perfect but I want to start somewhere

**Comment by trentmc** [comment]
I agree that ocean.py needs better documentation.

And I agree with having a few pages detailing some of the functionalities, to start somewhere.

And I agree to not import the generated documentation.

But: the title of this issue is "Generate docs from ocean-lib". That - along with the description and the PR -- appears contradictory with the previous point.

We already had this. It was pretty. It took a lot of work to get there. Why was it removed, only to replace with a super basic version?

Btw there's already a github issue for good docs for ocean.py: #1184. For the style of eth-brownie or zama, like I mentioned above. @AnaLoznianudrecronjetech and I discussed this a few months ago and it seems that was the direction. So, what's changed?

**Comment by AnaLoznianu** [comment]
Nothing has changed. We want to get to a point where we have good documentation for our components. As you can imagine this is a tone of work. 
Having this autogenerated content for the ocean.lib that we can reference from the docs.oceanprotocol.com is one of the steps

**Comment by trentmc** [comment]
If we auto-generate then we are repeating work of the past. Why would we repeat that?

And auto generated had big flaws. That's why it was removed. 

Then everyone agreed (iirc) that we should have actually-useful docs. I wrote #1184 to reflect this, which also incorporated my discussions with you @AnaLoznianu .

I propose:
- we close this issue as wontfix. And the associated pr
- and we do #1184. For docs that people might actually use.
